app.js

```diff
@@ -13,95 +15,139 @@ dotenv.config();
-    res.render('pages/register', {
-      title: 'register'
-    })
-  });
-
-  app.get('/add', profileForm);
-  app.post('/add', upload.single('photo'), add);
-  
-  //bucketlist
-  app.get('/bucketlist', showBucketlistOverview);
-  app.get('/bucketlistResults', showBucketlistResults);
-
-  app.post('/bucketlistResults', showBucketlistResults);
-  
-  
-  // If there is no page found give an error page as page
-  app.get('*', (req, res) => {
-    res.status(404).render('pages/404', {
-      url: req.url,
-      title: 'Error 404',
-    })
-  });
-  
-  
-    //function render bucketlistOverview page
-  function showBucketlistOverview(req, res) {
-    res.render('pages/bucketlist/bucketlistOverview', {
-      title: 'bucketlist'
+  .then(async client => {
+    let data = []
+    const db = client.db("dateApp");
+
+    data = await db.collection("users").find({}).toArray();
+
+    // Resolve GET request
+    app.get('/', (req, res) => {
+      res.render('pages/index', {
+        title: 'home'
+      })
     });
-  };
-
-  //function render bucketlistResultaat page
-  function showBucketlistResults(req, res) {
-    res.render('pages/bucketlist/bucketlistResults', {
-      title: 'bucketlistoverview'
-    }, {
-      interestView: data
+
+    app.get('/register', (req, res) => {
+      res.render('pages/register', {
+        title: 'register'
+      })
     });
-  };
-
-  function profileForm(req, res) {
-    res.render('add.ejs')
-  }
-
-  function add(req, res, next) {
-    db.collection('profiles').insertOne({
-      name: req.body.name,
-      photo: req.file ? req.file.filename : null,
-      age: req.body.age,
-      bio: req.body.bio
-    }, done);
-
-    function done(err, data) {
-      if (err) {
-        next(err)
-      } else {
-        res.redirect('/' + data.insertedId)
+
+    app.get('/add', profileForm);
+    app.post('/add', upload.single('photo'), add);
+
+    //bucketlist
+    app.get('/bucketlist', showBucketlistOverview);
+    app.get('/bucketlistResults', showBucketlistResults);
+    app.get('/bucketlistOverview', informatieShow);
+    app.get('/bucketlistOverview/:id', singleCountryInfo);
+
+    app.post('/bucketlistOverview', saveBucketlistResults);
+
+    function showBucketlistResults(req, res) {
+      res.render('pages/bucketlist/bucketlistResults', {
+        title: 'bucketlistResults'
+      });
+    };
+
+    //function render bucketlistOverview page
+    function showBucketlistOverview(req, res) {
+      res.render('pages/bucketlist/bucketlistOverview', {
+        title: 'bucketlist'
+      });
+    };
+
+    // save the form data to the database
+    function saveBucketlistResults(req, res) {
+      const country = new Country(req.body);
+
+      country.save()
+        .then((result) => {
+          res.redirect('bucketlistOverview')
+        })
+        .catch((err) => {
+          console.log(err);
+        })
+    };
+
+    //function to find the saved data en show
+    function informatieShow(req, res) {
```

Comment:
Maybe smarter and consistent to call a function in full english instead of dutch+english